### PR TITLE
feat: print call trees on error

### PR DESCRIPTION
**Pull request by charles-cooper**
### What I did
print call trees on error
fix some bugs (related to modules, printing errors)

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

**Review by DanielSchiavini — COMMENTED** [review]
cool

File: boa/contracts/abi/abi_contract.py [hunk 1]
```diff
@@ -290,7 +290,7 @@ def marshal_to_python(self, computation, abi_type: list[str]) -> tuple[Any, ...]
         try:
             return abi_decode(schema, computation.output)
         except ABIError as e:
-            raise BoaError(self.stack_trace(computation)) from e
+            raise BoaError(computation.call_trace, self.stack_trace(computation)) from e
```
[the remark this answers is not in the document]
Comment (charles-cooper) [reply boa/contracts/abi/abi_contract.py:293]: yea, good idea

File: boa/contracts/call_trace.py [hunk 2]
```diff
@@ -1,27 +1,47 @@
...
+from eth.abc import ComputationAPI
 
 from boa.rpc import json
 from boa.util.abi import Address, abi_decode
 
 
 class TraceSource:
-    def format(self, input: bytes, output: bytes):
-        return f"{self}{self._format_input(input)}{self.format_output(output)}"
+    def format(self, input_: bytes, is_error: bool, output: bytes):
```
[the remark this answers is not in the document]
Comment (charles-cooper) [reply boa/contracts/call_trace.py:14]: yea, i found both ways a little bit weird. let me fiddle with it

File: boa/contracts/base_evm_contract.py [hunk 3]
```diff
@@ -43,11 +43,11 @@ def call_trace(self) -> TraceFrame:
 
     def handle_error(self, computation):
         try:
-            raise BoaError(self.stack_trace(computation))
+            raise BoaError(computation.call_trace, self.stack_trace(computation))
         except BoaError as b:
             # modify the error so the traceback starts in userland.
             # inspired by answers in https://stackoverflow.com/q/1603940/
```
[the remark this answers is not in the document]
Comment (charles-cooper) [reply boa/contracts/base_evm_contract.py:49]: i think so, what do you mean?

[the remark this answers is not in the document]
Comment (DanielSchiavini) [reply boa/contracts/base_evm_contract.py:50]: I assumed the `from None` was taking care that the "traceback starts in userland"

[the remark this answers is not in the document]
Comment (charles-cooper) [reply boa/contracts/base_evm_contract.py:50]: ah no, it's the `strip_internal_frames` function which does that

**Pull request merged**